.dev/base-files/README.md

```diff
@@ -0,0 +1,258 @@
+* Perform any build tasks, such as running ``npm`` to package up React code.
+* Finally, use the Dockerfile to combine the various parts (source files, 
+  zbundle, npm-generated archives) together into a Docker image.
+
+### Run
+
+Once your Docker image has been built, you can the app as a container via
+``python -m ci container run``.  The app will start on http://127.0.0.1:8888.
+Note that other URLs or IP addresses may be printed to the console during
+app startup, and should be ignored.  Type Ctrl-C to exit.
+
+### Test
+
+You can run ``python -m ci container test`` to make sure your app provides the
+various interfaces expected by Edge.  This is a good check to reduce the risk
+that your app won't work when deployed to Edge.
+
+### Publish
+
+Running ``python -m ci container publish`` will push your app to the appropriate
+Docker registry.  Be sure you are logged in (``docker login <registry>``)
+before attempting.  **Before publishing, you should run preflight checks as
+described later in this document.**
+
+### Bundle rebuild
+
+During the Docker image build, a .zbundle file is constructed with dependencies.
+This is expensive to create, so it isn't regenerated each time the
+``container build`` command is run.  If your dependencies change, you should
+rebuild the bundle file by running ``python -m ci containers generate_bundle``.
+
+
+## Preflight tasks
+
+These should be used as a "final check" before publishing to Edge.  They will
+test your app against a local JupyterHub instance.
+
+Run ``python -m ci preflight run`` to run JupyterHub.  You can log in at
+http://127.0.0.1:8000.  Enter the username "edge" and the password "password".
+When finished, type Ctrl-C to exit.
+
+To run the preflight check in an automated way, do 
+``python -m ci preflight test``.  This can be e.g. incorporated into a 
+GitHub Actions workflow.
+
+
+## edge_settings.json
+
+You will notice an edge_settings.json.example file in this example's root.
+During local development, it is often convenient to have a genuine EdgeSession
+object for e.g. interacting with the Edge file system.  To make an EdgeSession
+available in your app during development, fill in the fields in this file
+and rename it to "edge_settings.json".  The "ci" module will pick up the
+information and set the appropriate environment variables when your app runs.
+
+**Please be careful not to check edge_settings.json into source control, and
+do not copy it into your Docker image.**  When running in the real Edge
+system, the EdgeSession object can be created automatically using the
+token, etc., of the user who launched the app.  This information is scraped
+from the environment when you call the ``EdgeSession()`` constructor.
+
+
+## Environment variables
+
+Within your app, you have access to a number of environment variables which
+may be useful for the development process:
+
+APP_VERSION: In local mode, this is a string with the Docker image name + tag.
```

Comment:
This is not built into all examples and is not supported by the default ContainerBuilder class. (The argument used to generate the default environment variable value would have to be passed in via Docker build command line argument, and it is not.)  It is unique to the original native flask app and is supplied as part of the containerization process rather than via the CI. It should either added to all or removed.

Adding:

- Add the required command line flag (`--build-arg CI_IMAGE={self.context.image}`) to ci.builders.ContainerBuilder.build
- Add the required Dockerfile arg and environment to every Dockerfile:
  ```
  ARG CI_IMAGE=quay.io/enthought/edge-native-app-flask-demo:latest
  ENV APP_VERSION=$CI_IMAGE
   ```

Removing:

- Remove from all Dockerfiles in examples
- Publish specific edge-native-base:0.9.0 and 1.0.0 images that hardcode the version values into text content in index.html
- Refactor edge-product end to end tests that verify version specific launches to use edge-native-base:0.9.0 and edge-native-base 1.0.0

Reply:
I am more in favor of having it removed because this feature exists at the example level/image rather than the base image and is more likely to be destroyed in the process of turning and example into a production application.

Reply:
I am happy to remove it entirely, and fix the fallout in the e2e test suite as needed (although I may need your help with that).